### PR TITLE
FEATURE: Spike content

### PR DESCRIPTION
**Pull request by lukehorvat**
@apj Is this looking alright so far?

### CODE REVIEW

**Comment by lukehorvat** [comment]
@apj Spike report done.

File: README.md [hunk 1]
```diff
@@ -1,3 +1,40 @@
...
-TODO.
+## Goals
+
+The goal of this spike was to assess the feasibility of using Node.js to programmatically interact with Git.
+
+## Technologies
+
+- [Git](https://git-scm.com)
+- [Node.js](https://nodejs.org)
+- [NPM](https://www.npmjs.com)
```
Comment (apj) [inline README.md]: @lukehorvat Include the Git `npm` package we used and versions of these things the spiked was carried out with.

File: README.md [hunk 2]
```diff
@@ -1,3 +1,40 @@
-# node-git-spike
+# Programmatic Git via Node.js
 
-TODO.
+## Goals
+
+The goal of this spike was to assess the feasibility of using Node.js to programmatically interact with Git.
```
Comment (apj) [inline README.md]: @lukehorvat Would be good to define what was feasibility means in this instance. This spike focused on assessing the feasibility of using a particular library for this purpose and so feasibility, to me, means did it work as intended and are there any practical issues to consider if we were to use it beyond the scope of this spike.

File: README.md [hunk 3]
```diff
@@ -1,3 +1,40 @@
...
+- [Git](https://git-scm.com)
+- [Node.js](https://nodejs.org)
+- [NPM](https://www.npmjs.com)
+
+## Tasks
+
+- Investigated various open-source Git libraries for Node.js.
+- Developed scripts to demonstrate basic programmatic usage of Git.
+
+## Outcomes
```
Comment (apj) [inline README.md]: @lukehorvat This isn't well represented in the examples Stan has, but it's good to focus on practical outcomes here, too. Having a couple of illustrative code snippets here shows, at a glimpse, how to use the library in this instance.

File: README.md (hunk 2 above)
Comment (apj) [inline README.md]: @lukehorvat To add to this, I'd say we want to evaluate whether the library is actually nice to use. It presents a maintenance problem for us if it's a piece of shit to understand and work with.

File: README.md [hunk 4]
```diff
@@ -1,3 +1,40 @@
...
+
+It appears to be quite easy to use Git programmatically from within a Node.js application, due to the wealth of Git libraries in the NPM registry. The [simple-git](https://github.com/steveukx/git-js) library was determined to be the best choice for basic Git interactions, and it worked relatively well for this spike.
```
Comment (apj) [inline README.md]: @lukehorvat Were other libraries considered? If so, why were they discarded?

File: README.md [hunk 5]
```diff
@@ -1,3 +1,40 @@
...
+
+## Issues / Risks
+
+N/A
+
+## Recommendations
+
+The simple-git library appears to be quite adequate for basic use cases. But for "complex" Git usage, one of the more feature-packed alternatives may be better.
```
Comment (apj) [inline README.md]: @lukehorvat List some of the popular feature-packed alternatives. What does "complex" mean in this case and why is it not suitable for our intended usage?

**Comment by lukehorvat** [comment]
@apj Better?

**Comment by apj** [comment]
@lukehorvat 👍

**Pull request merged**